lib/pbench/server/database/models/datasets.py

```diff
@@ -1140,21 +1170,17 @@ def get(dataset: Dataset, key: str) -> "Metadata":
             The Metadata model object
         """
         try:
-            meta = (
-                Database.db_session.query(Metadata)
-                .filter_by(dataset=dataset, key=key)
-                .first()
-            )
-        except SQLAlchemyError as e:
+            meta = __class__._query(dataset, key, user).first()
+        except SQLAlchemyError:
             Metadata.logger.exception("Can't get {}>>{} from DB", dataset, key)
-            raise MetadataSqlError("getting", dataset, key) from e
+            raise MetadataSqlError("getting", dataset, key)
```

Comment:
Peter pointed out that there is a subtle difference in the resulting output of the traceback which justifies including the `from` keyword.  I'm not sure that it's worth adding the `as` just so that we can have the `from`, but I felt obliged to wave that flag (since he and I spent a lot of mana wrangling over it).

Reply:
That's interesting since I don't believe I was a party to that discussion, and I know that my habit of dropping the `from` (designating, I see, an explicitly chained exception rather than an unintentional secondary exception during handling of a primary) came from your advocacy a while back.

But, yeah; I just (re-)read [PEP-3134](https://legacy.python.org/dev/peps/pep-3134/): I think I see the point, and I agree that `__cause__` would be a bit better than letting it default to `__context__` which implies to the traceback reader that the chain was unintended.

So, basically, we want to put back most of the `from` clauses ... 😬

Reply:
Yeah.  Sorta.  Not really, but...yeah.  😏